### ISSUE TITLE
coreos.ignition.v2_2.security.tls completes slowly or fails on qemu

### ISSUE THREAD

**Issue by bgilbert** [coreos/mantle#931]
`coreos.ignition.v2_2.security.tls` has recently started running slowly or failing on qemu.  We think this is an entropy problem: Linux has recently stopped allowing urandom to be read before the entropy pool has been initialized, and kola intentionally doesn't plumb entropy into the VM from the host OS.

Journal fragment from the client machine, in a case where the test eventually passed:

```
Oct 10 21:35:38.238272 ignition[382]: Adding "10.0.0.2" to list of CAs
Oct 10 21:35:38.239222 ignition[382]: GET https://10.0.0.2: attempt #1
Oct 10 21:35:38.841117 systemd-networkd[231]: eth0: Gained IPv6LL
Oct 10 21:35:40.696679 systemd-networkd[231]: eth0: Configured
Oct 10 21:38:35.613474 kernel: random: crng init done
Oct 10 21:38:35.613588 kernel: random: 7 urandom warning(s) missed due to ratelimiting
Oct 10 21:38:35.634955 ignition[382]: GET result: OK
Oct 10 21:38:35.636203 ignition[382]: fetched referenced config: {
                                              "ignition": { "version": "2.1.0" },
                                              "storage": {
                                                  "files": [{
                                                      "filesystem": "root",
                                                      "path": "/var/resource/data",
                                                      "contents": { "source": "data:,kola-data" }
                                                  }]
                                              }
                                          }
Oct 10 21:38:35.636947 ignition[382]: Adding "10.0.0.2" to list of CAs
Oct 10 21:38:35.638040 ignition[382]: Adding "10.0.0.2" to list of CAs
```

If this is indeed an entropy problem, it may help to hardcode the keypair in the test rather than generating it dynamically.

**Comment by arithx** [comment]
Capturing some notes from out of band discussions:

We're seeing the ratelimiting on the client (not the server where the keypairs are generated). This seems to be the result of the SSL/TLS handshake requiring entropy from both the client & server and as a result of the host's entropy not being used on the QEMU platform there is not enough available entropy inside the initramfs when requested by this handshake.

Still need to think through this more but one initial idea would be to add a flag that would allow a test to request additional entropy and QEMU clusters would be able to detect this and plumb through the host's entropy.

Another potential option would be to disable this test on low entropy platforms (QEMU, GCE).

**Comment by bgilbert** [comment]
Ignition requiring entropy before the system entropy pool is ready filed as https://github.com/coreos/ignition/issues/645.

On the server side, `TLSServe` could pass a `math/rand/Rand` instance in `tls.Config`.  It's not secure but that doesn't matter here.

**Comment by bgilbert** [comment]
Actually, the `TLSServe` change is unnecessary, since it runs long after entropy should be ready.

**Comment by arithx** [comment]
Closed in #936
